scripts/linux_containers_on_windows/generate_kernel_initrd.sh

```diff
@@ -2,31 +2,29 @@
 
 set -xe
 
-. utils.sh
-
 # root location of the already build kernel and opengcs artifacts
 KERNEL_ARTIFACTS_PATH="$1"
-OPENGCS_ARTIFACTS_PATH="$2"
 # locations of the extra packages that need to go into the initrd
-BUILD_EXTRA_ARTIFACTS="$3"
-BUILD_BASE_DIR="$4"
-KERNEL_VERSION="$5"
+BUILD_EXTRA_ARTIFACTS="$2"
+BUILD_BASE_DIR="$3"
+KERNEL_VERSION="$4"
+CLEAN_ENV="$5"
 
 
-BUILD_PATH="$BUILD_BASE_DIR/kernel-build-folder/initrd_build"
+BUILD_PATH="${BUILD_BASE_DIR}/kernel-build-folder/initrd_build"
 KERNEL_ARTIFACT_DIR_PATH=$(ls -td -- "$KERNEL_ARTIFACTS_PATH"/* |  grep  __msft-kernel_$KERNEL_VERSION | head -n 1)
-OPENGCS_ARTIFACT_DIR_PATH=$(ls -td -- $OPENGCS_ARTIFACTS_PATH/* |  grep __opengcs | head -n 1)
+OPENGCS_ARTIFACT_DIR_PATH=$(ls -td -- $KERNEL_ARTIFACTS_PATH/* |  grep __opengcs | head -n 1)
 
 function get_kernel_bits() {
     # here is where the initrd generate will take place, can't see any other reason to
     # do this somewhere else
-    if [ -d "$KERNEL_ARTIFACT_DIR_PATH" ]
+    if [[ -d "$KERNEL_ARTIFACT_DIR_PATH" ]]
     then
         pushd "$BUILD_PATH"
         vmlinuz_file="bootx64.efi"
-        sudo cp "$KERNEL_ARTIFACT_DIR_PATH/$vmlinuz_file" .
```

Comment:
applies to all sudo in the code:
never use sudo if you don't have to -- use fakeroot if possible